types/react-overlays/lib/Modal.d.ts

```diff
@@ -1,129 +1,131 @@
-   * A css class or classes for the backdrop component.
-   */
-  backdropClassName?: string;
-
-  /**
-   * A css class or set of classes applied to the modal container when the modal is open,
-   * and removed when it is closed.
-   */
-  containerClassName?: string;
-
-  /**
-   * Close the modal when escape key is pressed
-   */
-  keyboard?: boolean;
-
-  /**
-   * A `react-transition-group@2.0.0` `<Transition/>` component used
-   * to control animations for the dialog component.
-   */
-  transition?: React.ComponentType<TransitionProps>;
-
-  /**
-   * A `react-transition-group@2.0.0` `<Transition/>` component used
-   * to control animations for the backdrop components.
-   */
-  backdropTransition?: React.ComponentType<TransitionProps>;
-
-  /**
-   * When `true` The modal will automatically shift focus to itself when it opens, and
-   * replace it to the last focused element when it closes. This also
-   * works correctly with any Modal children that have the `autoFocus` prop.
-   *
-   * Generally this should never be set to `false` as it makes the Modal less
-   * accessible to assistive technologies, like screen readers.
-   */
-  autoFocus?: boolean;
-
-  /**
-   * When `true` The modal will prevent focus from leaving the Modal while open.
-   *
-   * Generally this should never be set to `false` as it makes the Modal less
-   * accessible to assistive technologies, like screen readers.
-   */
-  enforceFocus?: boolean;
-
-  /**
-   * When `true` The modal will restore focus to previously focused element once
-   * modal is hidden
-   */
-  restoreFocus?: boolean;
-
-  /**
-   * A ModalManager instance used to track and manage the state of open
-   * Modals. Useful when customizing how modals interact within a container
-   */
-  manager?: ModalManager;
+	className?: string;
+
+	/**
+	 * Set the visibility of the Modal
+	 */
+	show?: boolean;
+
+	/**
+	 * A Node, Component instance, or function that returns either. The Modal is appended to it's container element.
+	 *
+	 * For the sake of assistive technologies, the container should usually be the document body, so that the rest of the
+	 * page content can be placed behind a virtual backdrop as well as a visual one.
+	 */
+	container?: React.ReactNode | Function;
+
+	/**
+	 * A callback fired when the Modal is opening.
+	 */
+	onShow?: Function;
+
+	/**
+	 * A callback fired when either the backdrop is clicked, or the escape key is pressed.
+	 *
+	 * The `onHide` callback only signals intent from the Modal,
+	 * you must actually set the `show` prop to `false` for the Modal to close.
+	 */
+	onHide?: Function;
+
+	/**
+	 * Include a backdrop component.
+	 */
+	backdrop?: boolean | 'static';
+
+	/**
+	 * A function that returns the dialog component. Useful for custom
+	 * rendering. **Note:** the component should make sure to apply the provided ref.
+	 *
+	 * ```js
+	 *  renderDialog={props => <MyDialog {...props} />}
+	 * ```
+	 */
+	renderDialog?(props: any): React.ReactNode;
```

Comment:
Kind of a bummer to have `props: any` but I see this was how `renderBackdrop()` was already annotated.